compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/viewinterop/SwingInteropContainer.desktop.kt

```diff
@@ -142,6 +156,29 @@ internal class SwingInteropContainer(
         return interopComponents[group]
     }
 
+    private fun updateInteropComponentsOrder() {
+        val interopHolders = interopComponents.values.toList()
+        val countInteropBelowByHolder = interopHolders.associateWith {
+            countInteropComponentsBelow(it)
+        }
```

Comment:
It must be simplified. Currently it makes sorted look up over an entire tree for each iteration. We can iterate only once and remove extra sorting below

Reply:
The repositioning in the AWT container is an O(N^2) deal anyway, so it shouldn't matter, but ok.